File: monsterhunter/src/main/java/com/juancarlos/monsterhunter/services/ToolService.java
package com.juancarlos.monsterhunter.services;

import com.juancarlos.monsterhunter.models.dto.tool.ToolBaseDTO;

import java.util.List;

public interface ToolService {

    // Metodo lista de herramientas
    List<ToolBaseDTO> getToolList();
}

File: Frontend/MonsterHunter/src/app/services/tool.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { RequestService, RequestServiceOptions } from './request.service';
import { GetToolListResponse } from '../models/response/GetToolListResponse.model';

@Injectable({
  providedIn: 'root'
})
export class ToolService {

  // Inyección del servicio personalizado de peticiones HTTP
  constructor(private requestService: RequestService) { }

  // Método para obtener la lista de herramientas
  public getToolList(): Observable<GetToolListResponse> {
    let url = `${environment.tool}`;

    const requestServiceOptions: RequestServiceOptions = {
      url: url,
      method: 'get',
      responseType: 'json',
    };

    return this.requestService.request(requestServiceOptions, false, true);
  }
}